UPGRADE-API-1.12.md

```diff
@@ -225,3 +225,209 @@ The list of changed contexts:
+ +      $this->request = $this->requestFactory->index($this->section, $resource, $this->authorizationHeader, $this->getToken());
+
+        return $this->request($this->request);
+    }
+ ```
+
+The `public function buildUploadRequest(string $resource): void;` method has been removed, as it's replaced by methods in  `Sylius/Behat/Client/RequestBuilder`.
+Example change of usage in `Sylius/Behat/Context/Api/Admin/ManagingAdministratorsContext`:
+
+ ```diff
+ -       $this->client->buildUploadRequest(Resources::AVATAR_IMAGES);
+ -       $this->client->addParameter('owner', $this->iriConverter->getIriFromItem($administrator));
+ -       $this->client->addFile('file', new UploadedFile($this->minkParameters['files_path'] . $avatar, basename($avatar)));
+ -       $response = $this->client->upload();
+        
+ +       $builder = RequestBuilder::create(
+ +           sprintf('/api/v2/%s/%s', 'admin', Resources::AVATAR_IMAGES),
+ +           Request::METHOD_POST,
+ +       );
+ +       $builder->withHeader('CONTENT_TYPE', 'multipart/form-data');
+ +       $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+ +       $builder->withHeader('HTTP_Authorization', 'Bearer ' . $this->sharedStorage->get('token'));
+ +       $builder->withParameter('owner', $this->iriConverter->getIriFromItem($administrator));
+ +       $builder->withFile('file', new UploadedFile($this->minkParameters['files_path'] . $avatar, basename($avatar)));
+
+ +       $response = $this->client->request($builder->build());
+ ```
+
+As you can see the builder contains the methods that makes the request responsive for additional headers and parameters.
+
+We also changed the `request(RequestInterface $request): Response` method visibility to public.
+
+ ```diff
+ -    private function request(RequestInterface $request): Response
+ +    public function request(RequestInterface $request): Response
+ ```
+
+#### Defining Section and ContentType changed  
+
+Some of the private methods were removed from `Sylius/Behat/Client/RequestInterface`:
+
+ ```diff
+ -    private static function prepareSection(?string $section): string
+ -   {
+ -       if ($section === null) {
+ -           return '';
+ -       }
+ -
+ -       return $section . '/';
+ -   }
+
+ -   private static function resolveHttpMethod(string $method): string
+ -   {
+ -       return $method === HttpRequest::METHOD_PATCH ? 'application/merge-patch+json' : 'application/json';
+ -   }
+ ```
+
+The `prepareSection()` method is no more needed as we changed the way of injecting `$section` parameter, so adding `/` is no more needed.
+
+ ```diff
+ -        return new self(
+ -           sprintf('/api/v2/%s%s', self::prepareSection($section), $resource),
+ -           HttpRequest::METHOD_GET,
+ -           $headers
+ -       );
+        
+ +       $builder = RequestBuilder::create(
+ +           sprintf('/api/v2/%s/%s', $section, $resource),
+ +           HttpRequest::METHOD_GET,
+ +       );
+ ```
```

Comment:
To simplify the note, I would be for removing this information as it is about private methods, that shouldn't change anything in the end application:
```suggestion
```